src/basic/string-util.h

```diff
@@ -197,6 +196,18 @@ char* strextend_with_separator_internal(char **x, const char *separator, ...) _s
 int strextendf_with_separator(char **x, const char *separator, const char *format, ...) _printf_(3,4);
 #define strextendf(x, ...) strextendf_with_separator(x, NULL, __VA_ARGS__)
 
+#define strprepend_with_separator(x, separator, ...)                            \
+        ({                                                                      \
+                char **_p_ = ASSERT_PTR(x), *_s_;                               \
+                _s_ = strextend_with_separator_internal(NULL, (separator), __VA_ARGS__, empty_to_null(*_p_), NULL); \
+                if (_s_) {                                                      \
+                        free(*_p_);                                             \
+                        *_p_ = _s_;                                             \
```

Comment:
free_and_replace() ?

Reply:
That would invalidate `_s_`, while we still need it as return value.